### PR TITLE
If plugins return false when login, should mark as failure.

### PR DESCRIPTION
**Pull request by asika32764**
Move the return line into if block.

I found that if plugin return FALSE from `onUserBeforeLogin`, Application will still return true. This situation should mark as failure and goto `onUserLoginFailure`.

### CODE REVIEW

File: libraries/src/Application/CMSApplication.php [hunk 1]
```diff
@@ -855,9 +855,9 @@ public function login($credentials, $options = array())
 
 				// The user is successfully logged in. Run the after login events
 				$this->triggerEvent('onUserAfterLogin', array($options));
+
```
Comment (SharkyKZ) [inline libraries/src/Application/CMSApplication.php]:
```suggestion

```

**Comment by bembelimen** [comment]
Did you check what happens if one login works (e.g. the default Joomla! login) and another fails...? Is the user logged in after?

**Comment by asika32764** [comment]
Must test in a custom plugin or custom login code. This issue will only happened if a developer attempts to use `$app->login()` to make their own login flow in 3rd party components.

I'm still thinking how to attach simple code to reproduce this issue.

**Comment by ghost** [comment]
@asika32764 whats the state of this Pull Request?

**Comment by SharkyKZ** [comment]
To reproduce the issue in core:

Enable `System - User Log` plugin.
Try to login to administrator area with unauthorized user (should see `You do not have access to the Administrator section of this site.` message).
Check `administrator/logs/error.php` log. The login failure should be logged but it's not.

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on c6fccec978006b87356fb2a4974775593c85961a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22936">issues.joomla.org/tracker/joomla-cms/22936</a>.</sub>

**Comment by ghost** [comment]
@SharkyKZ thanks for the clear Test Instructions. At CloudAccess i have to upgrade to unlock Logs, so can't test.

**Comment by asika32764** [comment]
Thanks @SharkyKZ

I'm trying to create a plugin to reproduce this issue but your way is more simpler to test.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e5a7c2d34339d8959f8643aad219013de1945f31<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22936">issues.joomla.org/tracker/joomla-cms/22936</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22936">issues.joomla.org/tracker/joomla-cms/22936</a>.</sub>

**Comment by HLeithner** [comment]
thx

**Pull request merged**